packages/scripts/index.js

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env node
+
+const chalk = require('chalk');
+const fs = require('fs');
+const loadConfigFile = require('rollup/dist/loadConfigFile');
+const ora = require('ora');
+const path = require('path');
+const process = require('process');
+const rollup = require('rollup');
+
+const getRollupConfigPath = (file) => {
+  if (file && !fs.existsSync(path.resolve(path.join('.', file)))) {
+    // If the user explicitly provides a rollup config file and if it is not available, then throw an error and exit.
+    console.error(`Could not find ${path.join('.', file)}`);
+    process.exit(1);
+  }
+  // If user does't provide a rollup config file, then search for `rollup.config.js` in the project directory. If it is
+  // not found, then use `basePluginRollupConfig.js` which contains pretty reasonable defaults.
+  const projectRollupConfigFilePath = path.resolve(path.join('.', file || 'rollup.config.js'));
+  return fs.existsSync(projectRollupConfigFilePath)
+    ? projectRollupConfigFilePath
+    : path.resolve(path.join(__dirname), 'basePluginRollupConfig.js');
+};
+
+const OUTPUT_DIR_REGEX = /\/?(\w+)$/;
+
+const getOutputDir = (output) => {
+  const outputDirMatch = OUTPUT_DIR_REGEX.exec(output);
+  return outputDirMatch.length > 1 ? outputDirMatch[1] : output;
+};
+
+const startHandler = ({ file }) => {
+  process.env.ROLLUP_WATCH = true;
+
+  loadConfigFile(getRollupConfigPath(file))
+    .then(async ({ options, warnings }) => {
+      const spinners = {};
+      const watcher = rollup.watch(options);
+
+      warnings.flush();
+      watcher.on('event', (event) => {
+        switch (event.code) {
+          case 'BUNDLE_START':
+            event.output.forEach((output) => {
+              const outputDir = getOutputDir(output);
+              const spinner = ora({
+                text: chalk.blue.bold(`${event.input} -> ${outputDir}`),
+              });
+              spinners[`${event.input}-${output}`] = spinner;
+
+              spinner.start();
+            });
+            break;
+          case 'BUNDLE_END':
+            event.output.forEach((output) => {
+              const spinner = spinners[`${event.input}-${output}`];
+              if (spinner) {
+                const outputDir = getOutputDir(output);
+                spinner.succeed(chalk.green.bold(`${event.input} -> ${outputDir}`));
+                delete spinners[`${event.input}-${output}`];
+              }
+            });
+            break;
+          case 'ERROR':
+            Object.entries(spinners).forEach(([key, spinner]) => {
+              spinner.fail(event.error);
+              delete spinners[key];
+            });
+            break;
+        }
+      });
+
+      watcher.close();
+    })
+    .catch((err) => console.error(err));
+};
```

Comment:
I think this section is worth adding high level comments about what it does (i.e., I think it launches rollup in watch mode, listens for bundler events, and start/stop spinners in the console).

I might be getting too far ahead, but it might be worth thinking about what we will do in the monorepo world. 
 I suspect we'll be watching many modules (core/docker/titus/amazon) at the same time.  Have you given this thought yet?

Reply:
> I think this section is worth adding high level comments about what it does (i.e., I think it launches rollup in watch mode, listens for bundler events, and start/stop spinners in the console).

Yep, that's what it is doing here. The code is a little hard to read unfortunately. I'll add some comments here.

> I might be getting too far ahead, but it might be worth thinking about what we will do in the monorepo world.
I suspect we'll be watching many modules (core/docker/titus/amazon) at the same time. Have you given this thought yet?

Yep, that is the idea. We should be able to watch many modules get built in parallel. If we go the `vite` route, I've read that there's some out-of-the-box support for handling serving files from dependent packages in a mono-repo. I'm going to dig deeper into that after migrating all spinnaker packages.